networkit/cpp/Unittests-X.cpp

```diff
@@ -1,81 +1,58 @@
+/*
+ * Unittests-X.cpp
+ *
+ *  Created on: 2012
+ *      Author: Christian Staudt <christian.staudt@kit.edu>,
+ *              Henning Meyerhenke <henning.meyerhenke@kit.edu>,
+ *              Manuel Penschuck <networkit@manuel.jetzt>
+ */
+
 
 // includes
+#include <algorithm>
 #include <iostream>
 #include <utility>
-//#include <cfenv>	// floating point exceptions
-#include <stdexcept>
 
 // GoogleTest
-#ifndef NOGTEST
 #include <gtest/gtest.h>
-#endif
 
 // OpenMP
 #include <omp.h>
 
-// necessary for some reasons?
-#include <networkit/Globals.hpp>
-#include <networkit/ext/optionparser.hpp>
 #include <networkit/auxiliary/Log.hpp>
-#include <networkit/graph/Graph.hpp>
 #include <networkit/auxiliary/Parallelism.hpp>
-
-
-using namespace NetworKit;
-
-
-// *** Option Parser Configuration ***//
+#include <networkit/ext/optionparser.hpp>
 
 class Arg: public OptionParser::Arg {
-
-static OptionParser::ArgStatus Required(const OptionParser::Option& option, bool msg)
-{
-  if (option.arg != 0)
-    return OptionParser::ARG_OK;
-
-  if (msg) {
-	  std::cout << "Option '" << option << "' requires an argument" << std::endl;
-  }
-  return OptionParser::ARG_ILLEGAL;
-}
-
+    static OptionParser::ArgStatus Required(const OptionParser::Option& option, bool msg)
+    {
+      if (option.arg != 0)
+        return OptionParser::ARG_OK;
+
+      if (msg) {
+          std::cout << "Option '" << option << "' requires an argument" << std::endl;
+      }
+      return OptionParser::ARG_ILLEGAL;
+    }
 };
 
-// TODO: clean up obsolete parameters
-enum  optionIndex { UNKNOWN, HELP, LOGLEVEL, THREADS, TESTS, RUNNABLE, DEBUG, BENCHMARKS, FILTER };
-const OptionParser::Descriptor usage[] =
-{
-{UNKNOWN,	0,	"" ,	"",				OptionParser::Arg::None,	"Options:" },
-{HELP,		0,	"h",	"help",			OptionParser::Arg::None,	"  --help  \t Print usage and exit." },
-{LOGLEVEL,	0,	"",		"loglevel",		OptionParser::Arg::Required,"  --loglevel=<LEVEL>  \t set the log level" },
-{THREADS,	0,	"",		"threads",		OptionParser::Arg::Required,"  --threads=<NUM>  \t set the maximum number of threads" },
-{TESTS,		0,	"t",	"tests",		OptionParser::Arg::None,	"  --tests \t Run unit tests"},
-{RUNNABLE,	0,	"r",	"run",			OptionParser::Arg::None,	"  --run \t Run unit tests which don't use assertions"},
-{DEBUG,		0,	"d",	"debug",		OptionParser::Arg::None,	"  --debug \t Run tests to debug some algorithms"},
-{BENCHMARKS,0,	"b",	"benchmarks",	OptionParser::Arg::None,	"  --benchmarks \t Run benchmarks"},
-{FILTER,	0,	"f",	"gtest_filter",	OptionParser::Arg::Required,"  --gtest_filter=<FILTER_PATTERN> \t Run tests that match the filter pattern" },
-{UNKNOWN,	0,	"",		"",				OptionParser::Arg::None,	"\nExamples:\n TODO" },
-{0,			0,	0,		0,				0,							0}
+enum  optionIndex { UNKNOWN, HELP, LOGLEVEL, THREADS, TESTS, RUNNABLE, DEBUG, BENCHMARKS };
+const OptionParser::Descriptor usage[] = {
+    {UNKNOWN,	0,	"" ,	"",				OptionParser::Arg::None,	"Options:" },
```

Comment:
If we touch this, it might be a good idea to fix the alignment by replacing tabs by spaces.

Reply:
Yes, indeed.